apps/src/applab/applab.js

```diff
@@ -1250,13 +1250,17 @@ function onInterfaceModeChange(mode) {
  */
 Applab.setWorkspaceMode = function (mode) {
   var designWorkspace = document.getElementById('designWorkspace');
-  designWorkspace.style.display = (ApplabInterfaceMode.DESIGN === mode) ? 'block' : 'none';
+  if (designWorkspace) {
+    designWorkspace.style.display = (ApplabInterfaceMode.DESIGN === mode) ? 'block' : 'none';
+  }
 
   var codeWorkspaceWrapper = document.getElementById('codeWorkspaceWrapper');
   codeWorkspaceWrapper.style.display = (ApplabInterfaceMode.CODE === mode) ? 'block' : 'none';
 
   var dataWorkspaceWrapper = document.getElementById('dataWorkspaceWrapper');
-  dataWorkspaceWrapper.style.display = (ApplabInterfaceMode.DATA === mode) ? 'block' : 'none';
+  if (dataWorkspaceWrapper) {
+    dataWorkspaceWrapper.style.display = (ApplabInterfaceMode.DATA === mode) ? 'block' : 'none';
+  }
```

Comment:
I did a few more things to fully reactify hide/show for code and design
workspaces in applab. PTAL!

On Tue, Jun 28, 2016 at 3:22 PM, Brent Van Minnen <notifications@github.com>
wrote:

> In apps/src/applab/applab.js
> <https://github.com/code-dot-org/code-dot-org/pull/9217#discussion_r68855142>
> :
>
> >
> >    var codeWorkspaceWrapper = document.getElementById('codeWorkspaceWrapper');
> >    codeWorkspaceWrapper.style.display = (ApplabInterfaceMode.CODE === mode) ? 'block' : 'none';
> >
> >    var dataWorkspaceWrapper = document.getElementById('dataWorkspaceWrapper');
> > -  dataWorkspaceWrapper.style.display = (ApplabInterfaceMode.DATA === mode) ? 'block' : 'none';
> > +  if (dataWorkspaceWrapper) {
> > +    dataWorkspaceWrapper.style.display = (ApplabInterfaceMode.DATA === mode) ? 'block' : 'none';
> > +  }
>
> For dataWorkspaceWrapper, could we not just have CodeWorkspaceContainer
> connect to state.interfaceMode and then get rid of this non React
> toggling of the display?
>
> i.e.
>
> connect(state => ({
>   showData: state.pageConstants.hasDataMode && state.interfaceMode && state.interfaceMode === ApplabInterfaceMode.DATA,
>   ...
> });
>
> At which point in that guys render we can just do:
>
> {this.props.showData && <DataWorkspace/>
>
> and don't need to render a hidden div.
>
> I'm not sure if whether we can/can't do the same thing for designWorkspace
> since it's a ProtectedDiv.
>
> Ultimately it feels a little weird to me that we have this component
> CodeWorkspaceContainer used by multiple apps, that has children
> (data/design workspaces) used by only one of the entry points. It's also
> worth noting that Gamelab has a similar concept in it's animation tab,
> although I think it handles it differently.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/9217/files/c1fbc73a06b04e93c07c51c6da380677595b6a61#r68855142>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHoY5a_CHhXPHMUkF2ssVyizV94jPfdsks5qQZ6WgaJpZM4JAiJR>
> .
>